src/MongoDB.Driver.Core/Core/ConnectionPools/ExclusiveConnectionPool.cs

```diff
@@ -140,12 +148,12 @@ public int UsedCount
         // public methods
         public IConnectionHandle AcquireConnection(CancellationToken cancellationToken)
         {
-            ThrowIfNotOpen();
-
             var helper = new AcquireConnectionHelper(this);
             try
             {
                 helper.CheckingOutConnection();
+                ThrowIfNotOpen();
```

Comment:
Just to confirm:
The "old" sequence of execution was essentially: ThrowIfDisposed, EnterWaitQueue, emit "CheckingOutConnection" event.
The "new" sequence of execution is: Emit "CheckingOutConnection" event, ThrowIfDisposed, EnterWaitQueue, 

Would changing this sequence of execution potentially be considered a breaking API change? Is there any scenario in which an application could rely on the old behavior?
@rstam

Reply:
Agreed, we have to make this change. I was merely wondering with what version we'd release this change. After some additional research, I think the order in which we raise events/throw exceptions is an implementation detail and since we have no evidence to believe that changing this order would break a large # of users, I'm fine with releasing this in a patch release.  Resolving this thread!